src/amoc_api_scenarios_handler.erl

```diff
@@ -10,112 +10,130 @@
          allowed_methods/2,
          content_types_provided/2,
          content_types_accepted/2,
-         process_request/2,
-         process_json/2]).
+         to_json/2,
+         from_json/2]).
 
--record(state, {action}).
+
+-record(state, {module_name, module_source}).
```

Comment:
It doesn't make sense using #state{} in this module. We are not keeping track of any data between requests. Please fix that